peer/peerwg/peermessage.go

```diff
@@ -10,35 +10,35 @@ import (
 	"tailscale.com/types/key"
 )
 
-const peerMessageSeparator byte = '\n'
+const handshakeSeparator byte = '\n'
 
-// WireguardPeerMessage is a message received from a wireguard peer, indicating
+// Handshake is a message received from a wireguard peer, indicating
 // it would like to connect.
-type WireguardPeerMessage struct {
+type Handshake struct {
 	// Recipient is the uuid of the agent that the message was intended for.
 	Recipient uuid.UUID `json:"recipient"`
-	// Disco is the disco public key of the peer.
-	Disco key.DiscoPublic `json:"disco"`
-	// Public is the public key of the peer.
-	Public key.NodePublic `json:"public"`
+	// DiscoPublicKey is the disco public key of the peer.
+	DiscoPublicKey key.DiscoPublic `json:"disco"`
+	// NodePublicKey is the public key of the peer.
+	NodePublicKey key.NodePublic `json:"public"`
 	// IPv6 is the IPv6 address of the peer.
 	IPv6 netaddr.IP `json:"ipv6"`
 }
 
-// WireguardPeerMessageRecipientHint parses the first part of a serialized
-// WireguardPeerMessage to quickly determine if the message is meant for the
-// provided agentID.
-func WireguardPeerMessageRecipientHint(agentID []byte, msg []byte) (bool, error) {
-	idx := bytes.Index(msg, []byte{peerMessageSeparator})
+// HandshakeRecipientHint parses the first part of a serialized
+// Handshake to quickly determine if the message is meant for the
+// provided recipient.
+func HandshakeRecipientHint(agentID []byte, msg []byte) (bool, error) {
+	idx := bytes.Index(msg, []byte{handshakeSeparator})
 	if idx == -1 {
 		return false, xerrors.Errorf("invalid peer message, no separator")
 	}
 
 	return bytes.Equal(agentID, msg[:idx]), nil
 }
 
-func (pm *WireguardPeerMessage) UnmarshalText(text []byte) error {
-	sp := bytes.Split(text, []byte{peerMessageSeparator})
+func (pm *Handshake) UnmarshalText(text []byte) error {
```

Comment:
should rename these method receivers too

Reply:
true will do